File: Frontend/safal_yatra/lib/app/models/getUserProfile.dart
// To parse this JSON data, do
//
//     final userProfileResponse = userProfileResponseFromJson(jsonString);

import 'dart:convert';

UserProfileResponse userProfileResponseFromJson(String str) =>
    UserProfileResponse.fromJson(json.decode(str));

String userProfileResponseToJson(UserProfileResponse data) =>
    json.encode(data.toJson());

class UserProfileResponse {
  final bool? success;
  final String? message;
  final Users? users;

  UserProfileResponse({
    this.success,
    this.message,
    this.users,
  });

  factory UserProfileResponse.fromJson(Map<String, dynamic> json) =>
      UserProfileResponse(
        success: json["success"],
        message: json["message"],
        users: json["users"] == null ? null : Users.fromJson(json["users"]),
      );

  Map<String, dynamic> toJson() => {
        "success": success,
        "message": message,
        "users": users?.toJson(),
      };
}

class Users {
  final String? fullName;
  final String? phoneNumber;
  final String? email;
  final String? address;
  final String? gender;
  final dynamic imageUrl;
  final DateTime? addedDate;

  Users({
    this.fullName,
    this.phoneNumber,
    this.email,
    this.address,
    this.gender,
    this.imageUrl,
    this.addedDate,
  });

  factory Users.fromJson(Map<String, dynamic> json) => Users(
        fullName: json["full_name"],
        phoneNumber: json["phone_number"],
        email: json["email"],
        address: json["address"],
        gender: json["gender"],
        imageUrl: json["image_url"],
        addedDate: json["added_date"] == null
            ? null
            : DateTime.parse(json["added_date"]),
      );

  Map<String, dynamic> toJson() => {
        "full_name": fullName,
        "phone_number": phoneNumber,
        "email": email,
        "address": address,
        "gender": gender,
        "image_url": imageUrl,
        "added_date": addedDate?.toIso8601String(),
      };
}

File: Backend/SafalYatra/user/getUserProfile.php
<?php
include "../config/connection.php";
include "../utils/authHelper.php";


if (!isset($_POST['token'])) {
    echo json_encode(array(
        "success" => false,
        "message" => "Token is required"
    ));
    die();
}

$token = $_POST['token']; // Storing the token from POST request

$userId = getUserId($token);

if (!$userId) {
    echo json_encode(array(
        "success" => false,
        "message" => "Invalid token"
    ));
    die();
}


$sql = "SELECT full_name, phone_number, email, address, gender, image_url, added_date FROM users WHERE user_id = $userId";
$result = mysqli_query($connect, $sql);

if (!$result) {
    echo json_encode(array(
        "success" => false,
        "message" => "Failed to fetch users data",

    ));

    die();
}
$user = mysqli_fetch_assoc($result);
echo json_encode(array(
    "success" => true,
    "message" => "Users data fetched successfully",
    "users" => $user

));
